### PR TITLE
Added `applyEither` for validation-style combination

### PR DESCRIPTION
**Pull request by reactormonk**
Used by https://github.com/justinwoo/purescript-simple-json/pull/56/files

### CODE REVIEW

**Comment by reactormonk** [comment]
The use-case here is a different combination logic once. For the same code, you'd need to convert to validation and back. This would be possible, but require either `either` or `validation` to depend on each other or even a third package just for `Either -> Validation` and back.

**Comment by hdgarrood** [comment]
`validation` already depends on `either` though; the definition of `V` is
```purescript
newtype V err result = V (Either err result)
```
and although the constructor isn't exported, you can freely convert back and forth with `either invalid pure` and `toEither`.

**Comment by reactormonk** [comment]
So to use this specific applicative combinator, you suggest to a) include purescript-validation b) write
`toEither $ apply (either invalid pure e1) (either invalid pure e2)`? That does sound like a whole lot of work.

**Comment by hdgarrood** [comment]
Sorry, `under V apply` doesn't typecheck. But with `V` and `un V` (which we will have if we export the constructor) I think it is easy enough to convert between them. But also, if you want this function, you probably should just be using `V` to begin with.

**Comment by hdgarrood** [comment]
Another thing we could do to make this more comfortable would be https://github.com/purescript/purescript-parallel/issues/26

**Comment by reactormonk** [comment]
tl;dr: not possible: https://github.com/purescript/purescript-parallel/pull/27

**Comment by hdgarrood** [comment]
I don't want to merge this, sorry. I'm concerned that people might expect it to behave in the same way as `apply` for Either, and I also think that it doesn't really make sense to provide this function other than in an actual Apply instance (as `validation` already does). If you do want this specific function, as opposed to the version which works with `V`, I think that's too much of a niche use case to deserve a spot in Data.Either, and it's easy enough to define inline, as you've done.
